app/src/main/java/com/kickstarter/viewmodels/MessageThreadHolderViewModel.java

```diff
@@ -88,7 +99,36 @@ public ViewModel(final @NonNull Environment environment) {
       this.unreadCountTextViewText = this.messageThread
         .map(MessageThread::unreadMessagesCount)
         .map(NumberUtils::format);
-      this.unreadIndicatorViewHidden = this.messageThread.map(m -> m.unreadMessagesCount() == 0);
+
+      this.unreadIndicatorViewHidden = hasUnreadMessages.map(BooleanUtils::negate);
+
+      this.messageThread
+        .compose(takeWhen(this.messageThreadCardViewClicked))
+        .subscribe(thread -> markedAsRead(thread, this.sharedPreferences));
+    }
+
+    private static @NonNull String cacheKey(final @NonNull MessageThread messageThread) {
+      return SharedPreferenceKey.MESSAGE_THREAD_HAS_UNREAD_MESSAGES + messageThread.id();
+    }
+
+    private static boolean hasUnreadMessages(final @NonNull MessageThread messageThread,
+      final @NonNull SharedPreferences sharedPreferences) {
+      return sharedPreferences.getBoolean(cacheKey(messageThread), messageThread.unreadMessagesCount() > 0);
+    }
+
+    private static void markedAsRead(final @NonNull MessageThread messageThread,
+      final @NonNull SharedPreferences sharedPreferences) {
+      final SharedPreferences.Editor editor = sharedPreferences.edit();
+      editor.putBoolean(cacheKey(messageThread), false);
+      editor.apply();
+    }
+
+    private static void setHasUnreadMessagesPreference(final @NonNull MessageThread messageThread,
+      final @NonNull SharedPreferences sharedPreferences) {
+
+      final SharedPreferences.Editor editor = sharedPreferences.edit();
+      editor.putBoolean(cacheKey(messageThread), messageThread.unreadMessagesCount() > 0);
+      editor.apply();
```

Comment:
this is a very good question, i was looking at the way we do `sharedPreferences` in our `ProjectViewModel` to store ref tag cookies and figured it'd be fine to not make a new preference in the `ApplicationModule`, but seems like we do have a `provideCookieManager` and thus we should add this to the ApplicationModule to make sure it's provided correctly

good catch!!! 🐦